src/main/java/com/learntechfree/ltf/article/ArticleService.java

```diff
@@ -12,15 +13,18 @@
     @Autowired
     private ArticleRepository articleRepository;
 
-    public Article createArticle(Article article){
+    public Article createArticle(Article article) {
+        
         return articleRepository.saveAndFlush(article);
     }
 
-    public List<Article> listArticles(int count){
+    public List<Article> listArticles(int count) {
+
         return articleRepository.findAll().stream().limit(count).collect(Collectors.toList());
     }
 
-    public Optional<Article> getArticle(Long id){
+    public Optional<Article> getArticle(Long id) {
+
         return articleRepository.findById(id);
     }
```

Comment:
So, here always make sure we are returning data when some other methods invoke this service. If `repository` is not sure about data (ie. it will return Optional) we should handle the exception right away. The service here looks like

```java
public Article getArticle(Long id) {
        return articleRepository.findById(id).orElseThrow(()-> new ResourceNotFoundException("Article","id",id));
}
```

And the ResourceNotFoundException looks like 
```java
@ResponseStatus(HttpStatus.NOT_FOUND)
public class ResourceNotFoundException extends RuntimeException {
    private final String resourceName;
    private final String fieldName;
    private final transient Object fieldValue;

    public ResourceNotFoundException(String resourceName, String fieldName, Object fieldValue) {
        super(String.format("%s not found with %s : '%s'", resourceName, fieldName, fieldValue));
        this.resourceName = resourceName;
        this.fieldName = fieldName;
        this.fieldValue = fieldValue;
    }
}
```

Reply:
Great let's do it then

Reply:
@akasrai  applied but still, ArticleResolver is not used so far.